packages/core/core/src/services/document-service/transform/utils.ts

```diff
@@ -1,9 +1,67 @@
-const switchIdForDocumentId = (output: Record<string, any>) => {
+import { Common } from '@strapi/types';
+import { IdMap } from './id-map';
+
+export const switchIdForDocumentId = (output: Record<string, any>) => {
   // Mutating for performance reasons
   const documentId = output.documentId;
   delete output.documentId;
   output.id = documentId;
   return output;
 };
 
-export { switchIdForDocumentId };
+export const transformFieldsOrSort = (input: string[] | string): string[] | string => {
+  if (Array.isArray(input)) {
+    return input.map((item) => (item === 'id' ? 'documentId' : item));
+  }
+  return input.replace('id', 'documentId');
+};
+
+export const transformFiltersOrPopulate = (
+  idMap: IdMap,
+  obj: Record<string, any>,
+  opts: { uid: Common.UID.Schema; locale?: string | null; isDraft: boolean }
+): Record<string, any> => {
+  const transformedObj: Record<string, any> = {};
+
+  for (const key of Object.keys(obj)) {
+    if (key === 'id') {
+      transformedObj.documentId = obj[key];
+    } else if (['filters', 'populate'].includes(key)) {
```

Comment:
I've been trying to rework this to use traverseEntity but am finding it doesn't support this case

As I mentioned on slack:

I think the issue is this line
https://github.com/strapi/strapi/blob/f0318206965826d295f5cf1a4f8473e7f7eaedba/packages/core/utils/src/traverse-entity.ts#L84

And situations like this conversion
      {
          input: { category: { filters: { id: 'documentId' } } },
          expected: { category: { filters: { documentId: 'documentId' } } },
        },
e.g. filters is not an attribute of the schema so my traversal exits and the logic doesn’t apply

I wanted to check that we definitely need to support the kind of input and output above. and if so wdyt the best way is? Is it simpler to just to do the conversion as I have in the PR atm?